src/fsharp/pars.fsy

```diff
@@ -5328,8 +5328,13 @@ opName:
   | LPAREN error rparen  
      {  reportParseErrorAt (lhs parseState) (FSComp.SR.parsErrorParsingAsOperatorName()); ident(CompileOpName "****", rhs parseState 2) }
 
+  /* (*) is a special case, since we need to distinguish it from the comments, so we compensate the position of operator itself, since parens are getting captured here as well, we do it so things like nameof(*) are demangled properly to '*' instead of op_Multiply */
   | LPAREN_STAR_RPAREN
-     {  ident(CompileOpName "*", rhs parseState 1) }
+     {  let m = rhs parseState 1
```

Comment:
@dsyme I hope it is a normal approach for it - just to create a new range for it from the existing one.